pyiron/lammps/control.py

```diff
@@ -201,42 +201,57 @@ def pressure_to_lammps(self, pressure, rotation_matrix):
         """
 
-        # Get something with six elements
-        if not hasattr(pressure, "__len__"):
-            if not np.isclose(np.matrix.trace(rotation_matrix), 3) and pressure == 0.0:
-                pressure = np.append(pressure * np.ones(3), pressure * np.ones(3))
-            else:
-                pressure = np.append(pressure * np.ones(3), 3 * [None])
-        else:
-            if len(pressure) > 6:
-                raise ValueError("Pressure can have a maximum of 6 values, (x, y, z, xy, xz, and yz), but got " +
-                                 "{}".format(len(pressure)))
-            all_pressures = 6 * [None]
-            all_pressures[:len(pressure)] = pressure
-            pressure = np.array(all_pressures, dtype=float)
-
-        # Cell degrees of freedom can be kept fixed using None for the pressure, check where that's done.
-        not_none_mask = (pressure!=None)
-        if not np.any(not_none_mask):
+        # If pressure is a scalar, only unit conversion is needed.
+        if np.isscalar(pressure):
+            return float(pressure) * LAMMPS_UNIT_CONVERSIONS[self["units"]]["pressure"]
+
+        # Normalize pressure to a list of 6 entries.
+        if len(pressure) > 6:
+            raise ValueError("Pressure can have a maximum of 6 values, (x, y, z, xy, xz, and yz), but got " +
+                             "{}".format(len(pressure)))
+        pressure = list(pressure) + (6 - len(pressure)) * [None]
+
+        if all(p is None for p in pressure):
             raise ValueError("Pressure cannot have a length but all be None")
 
-        # If necessary, rotate the pressure tensor to the Lammps coordinate frame
-        if not np.isclose(np.matrix.trace(rotation_matrix), 3):
-            if not np.all(not_none_mask):
-                raise ValueError("Cells which are not orthorhombic or rectangular are incompatible with Lammps "
-                                 "constant pressure calculations unless the entire pressure tensor is defined."
+        # If necessary, rotate the pressure tensor to the Lammps coordinate frame.
+        # Isotropic, hydrostatic pressures are rotation invariant.
+        is_isotropic_hydrostatic = (
+            # All diagonal components must be equal and not None.
+            pressure[0] is not None and pressure[1] is not None and pressure[2] is not None
+            and np.isclose(pressure[0], pressure[1]) and np.isclose(pressure[1], pressure[2])
+            # Either all shear components must be None, or all shear components must be zero.
+            and (
+                (pressure[3] is None and pressure[4] is None and pressure[5] is None)
+                or
+                (pressure[3] is not None and np.isclose(pressure[3], 0.0) and
+                 pressure[4] is not None and np.isclose(pressure[4], 0.0) and
+                 pressure[5] is not None and np.isclose(pressure[5], 0.0))
+            )
+        )
```

Comment:
I do allow `pressure=[p,p,p,None,None,None]` and `pressure=[p,p,p,0,0,0]` for now with rotation ≠ I and skip the rotation. I'm not too attached to this, could be removed if nobody likes it.

Reply:
It's a pain-in-the-butt workaround on the backend, but I think it's nice to supply that behaviour on the frontend so I'm in favour.

Reply:
At least this part: `np.isclose(pressure[0], pressure[1]) and np.isclose(pressure[1], pressure[2])` could be `np.isclose(np.var(pressure[:3]), 0)`

Reply:
And this `pressure[3] is None and pressure[4] is None and pressure[5] is None` can be `np.all(pressure[3:]==None)` (I hope `pressure` is a numpy array)

Reply:
`pressure[0] is not None and pressure[1] is not None and pressure[2] is not None` -> `np.all(pressure[:3] != None)`